README.md

```diff
@@ -235,58 +235,69 @@ For headaches, take Ibuprofen
+The version "latest" matches the latest available tagged version for the module
+containing the main package. gobin will make a network request to resolve
+"latest" unless -nonet is provided, in which case it will attempt to resolve
+"latest" within the GOPATH module download cache. Versions that take the form
+of a revision identifier, e.g. a branch name, can only be resolved with a
+network request and hence are incompatible with -nonet.
 
-In global mode, gobin installs the main packages to the directories
-gobin/$module@$version/$main_pkg under your user cache directory. See the
-documentation for os.UserCacheDir for OS-specific details on how to configure
-its location.
-
-In main-module mode, gobin installs the main packages to the directories
-.gobincache/$module@$version/$main_pkg under the directory containing the main
-module's go.mod.
+If no version is specified for a main package, gobin behaves differently in
+global and main-module modes. In global mode, gobin attempts to resolve the
+latest tagged version available in the module download cache. In main-module
+module, gobin attempts to resolve the current version via the main module's
+go.mod. If this resolution fails in either mode, "latest" is assumed as the
+version.
 
 By default, gobin installs the main packages to $GOBIN (or $GOPATH/bin if GOBIN
 is not set, which defaults to $HOME/go/bin if GOPATH is not set).
 
-The -r flag takes exactly one main package argument and runs that package.  It
-is similar therefore to go run. Any arguments after the single main package
+The -run flag takes exactly one main package argument and runs that package.
+It is similar therefore to go run. Any arguments after the single main package
 will be passed to the main package as command line arguments.
 
 The -p flag prints the cache install path for each of the provided packages
 once versions have been resolved.
 
-The -r and -p flags are mutually exclusive.
+The -run and -p flags are mutually exclusive.
 
-The -g flag forces gobin to perform a network fetch for the provided main
-packages.
+The -u flag forces gobin to check for the latest tagged version. As such, it
+has no effect if any form of version is specified.
 
-The -n flag prevents network access and instead uses the GOPATH module download
-cache where required.
+The -nonet flag prevents network access and instead uses the GOPATH module
+download cache where required for version resolution, including "latest".
 
-The -g and -n flags are mutually exclusive.
+The -u and -nonet flags are mutually exclusive.
 
 The -m flag causes gobin to use the main module (the module containing the
 directory where the gobin command is run). The main module is given by go env
 GOMOD. Without this flag gobin effectively runs as a "global" tool.
 
 It is an error for a non-main package to be provided as a package argument.
 
+
+Cache directories
+=================
+
+In global mode, gobin installs the main packages to the directories
```

Comment:
With respect to my suggestions about "global mode" above, this could be:

    Unless the `-m` flag is specified, ... etc

Reply:
Agreed.